src/components/VisualTesterKeymap.vue

```diff
@@ -0,0 +1,273 @@
+import BaseKeymap from '@/components/BaseKeymap';
+import TesterKey from '@/components/TesterKey';
+
+export default {
+  name: 'visual-tester-keymap',
+  extends: BaseKeymap,
+  async mounted() {
+    document.addEventListener('keydown', this.keydown);
+    document.addEventListener('keyup', this.keyup);
+    await this.init();
+    this.setSize(this.calculateMax(this.layout));
+  },
+  beforeDestroy() {
+    document.removeEventListener('keydown', this.keydown);
+    document.removeEventListener('keyup', this.keyup);
+  },
+  computed: {
+    ...mapState('tester', [
+      'layout',
+      'defaults',
+      'layouts',
+      'config',
+      'keymap',
+      'codeToPosition'
+    ]),
+    ...mapGetters('keymap', ['colorway']),
+    styles() {
+      let styles = [];
+      styles.push(`width: ${this.width}px;`);
+      styles.push(`height: ${this.height}px;`);
+      styles.push(`font-size: ${this.fontsize * this.config.SCALE}em;`);
+      return styles.join('');
+    },
+    testerLayer() {
+      if (this.keymap.length === 0) {
+        return [];
+      }
+      const layout = this.layouts[this.layout];
+      const keymap = this.keymap[0];
+      // Calculate Max with given layout
+      // eslint-disable-next-line no-console
+      this.profile && console.time('currentLayer');
+      let curLayer = layout.map((pos, index) => {
+        let _pos = Object.assign({ w: 1, h: 1 }, pos);
+        const coor = this.calcKeyKeymapPos(_pos.x, _pos.y);
+        const dims = this.calcKeyKeymapDims(_pos.w, _pos.h);
+        return Object.assign(
+          {
+            id: index,
+            layer: 0,
+            meta: keymap[index]
+          },
+          coor,
+          dims
+        );
+      });
+      // eslint-disable-next-line no-console
+      this.profile && console.timeEnd('currentLayer');
+      return curLayer;
+    },
+    displayKeyCode() {
+      return this.displayHex
+        ? `0x${this.lastKeyCode.toString(16)}`
+        : this.lastKeyCode;
+    }
+  },
+  methods: {
+    ...mapMutations('keymap', ['resizeConfig']),
+    ...mapMutations('tester', ['setActive', 'setDetected']),
+    ...mapActions('tester', ['init']),
+    getComponent() {
+      return TesterKey;
+    },
+    keyup(ev) {
+      const endTS = performance.now();
+      const evStr = this.formatKeyEvent(ev, endTS);
+      ev.preventDefault();
+      ev.stopPropagation();
+      const pos = this.codeToPosition[this.firefoxKeys(ev.code)];
+      this.writeToStatus(`KEY-UP ---- QMK: '${this.getQMKCode(pos)}' ${evStr}`);
+      if (!isUndefined(pos)) {
+        this.setDetected(pos);
+      }
+    },
+    keydown(ev) {
+      if (ev.repeat) {
+        return;
+      }
+      this.timing[ev.code] = performance.now();
+      ev.preventDefault();
+      ev.stopPropagation();
+      const pos = this.codeToPosition[this.firefoxKeys(ev.code)];
+      this.writeToStatus(
+        `KEY-DOWN -- QMK: '${this.getQMKCode(pos)}' ${this.formatKeyEvent(ev)}`
```

Comment:
This can be exported in a `formatPrompt` function don't you think?

Reply:
yeah, good idea. Sometimes I don't do this until I hit the 3rd use.

Reply:
I looked into it and they're different enough it doesn't really help, it just makes it harder to figure out what's going on, so I'm going to leave it for now.